src/io/flutter/view/FlutterView.java

```diff
@@ -373,6 +383,109 @@ public void debugActive(@NotNull FlutterViewMessages.FlutterDebugEvent event) {
+    final JxBrowserManager jxBrowserManager = JxBrowserManager.getInstance();
+    final DevToolsManager devToolsManager = DevToolsManager.getInstance(app.getProject());
+
+    presentLabel(toolWindow, INSTALLATION_IN_PROGRESS_LABEL);
+
+    // Start devtools while waiting for JxBrowser download.
+    devToolsManager.installDevTools();
+
+    if (jxBrowserManager.getStatus().equals(JxBrowserStatus.INSTALLED)) {
+      handleJxBrowserInstalled(app, inspectorService, toolWindow);
+    } else {
+      startJxBrowserInstallationWaitingThread(app, inspectorService, toolWindow);
+    }
+  }
+
+  protected void startJxBrowserInstallationWaitingThread(FlutterApp app, InspectorService inspectorService, ToolWindow toolWindow) {
+    ApplicationManager.getApplication().executeOnPooledThread(() -> {
+      waitForJxBrowserInstallation(app, inspectorService, toolWindow);
+    });
+  }
+
+  protected void waitForJxBrowserInstallation(FlutterApp app, InspectorService inspectorService, ToolWindow toolWindow) {
+    final JxBrowserManager jxBrowserManager = JxBrowserManager.getInstance();
+    // Periodically check for download to finish.
+    // TODO(helin24): Maybe set up JxBrowserManager as listener instead.
+    int waitSeconds = 1;
+    while (waitSeconds <= 128) {
+      final JxBrowserStatus newStatus = jxBrowserManager.getStatus();
+      if (newStatus.equals(JxBrowserStatus.INSTALLED)) {
+        handleJxBrowserInstalled(app, inspectorService, toolWindow);
+        return;
+      } else if (newStatus.equals(JxBrowserStatus.INSTALLATION_FAILED)) {
+        handleJxBrowserInstallationFailed(app, inspectorService, toolWindow);
+        return;
+      } else {
+        ThreadUtil.sleep(waitSeconds);
+        waitSeconds = waitSeconds * 2;
+      }
+    }
+    // TODO(helin24): Are there better options for this case? e.g. stop installation and retry, link to open in browser?
+    presentLabel(toolWindow, INSTALLATION_TIMED_OUT_LABEL);
+  }
+
+  protected void handleJxBrowserInstallationFailed(FlutterApp app, InspectorService inspectorService, ToolWindow toolWindow) {
+    // Allow user to manually restart.
+    presentClickableLabel(toolWindow, "JxBrowser installation failed. Click to retry.", new MouseAdapter() {
+      @Override
+      public void mouseClicked(MouseEvent e) {
+        JxBrowserManager.getInstance().retryFromFailed(app.getProject());
+        handleJxBrowserInstallationInProgress(app, inspectorService, toolWindow);
+
+      }
+    });
+  }
+
+  protected void presentLabel(ToolWindow toolWindow, String text) {
+    final JBLabel label = new JBLabel(text, SwingConstants.CENTER);
+    label.setForeground(UIUtil.getLabelDisabledForeground());
+    replacePanelLabel(toolWindow, label);
+  }
+
+  protected void presentClickableLabel(ToolWindow toolWindow, String text, MouseAdapter mouseAdapter) {
+    final JBLabel label = new JBLabel(text, SwingConstants.CENTER);
```

Comment:
I believe there is a JBLink or similar class that is already styled to look like a link. I think the performance tab sohws an example of using it.